src/test/v2.0/advanced/intent-support-2.0.ts

```diff
@@ -0,0 +1,286 @@
+
+declare let fdc3: DesktopAgent;
+const raiseIntentDocs = "\r\nDocumentation: " + APIDocumentation2_0.raiseIntent + "\r\nCause";
+
+export class RaiseIntentControl2_0 {
+  async receiveContext(contextType: string, waitTime?: number, expectedId?: string): Promise<Context> {
+    let timeout;
+    const appControlChannel = await getOrCreateChannel("app-control");
+    const messageReceived = new Promise<Context>(async (resolve, reject) => {
+      const listener = await appControlChannel.addContextListener(contextType, (context) => {
+        if (expectedId) {
+          expect(context.id).to.be.equal(expectedId);
+        }
+        resolve(context);
+        clearTimeout(timeout);
+        listener.unsubscribe();
+      });
+
+      //if no context received reject promise
+      const { promise: sleepPromise, timeout: theTimeout } = sleep(waitTime ?? undefined);
+      timeout = theTimeout;
+      await sleepPromise;
+      reject("No context received from app B");
+    });
+
+    return messageReceived;
+  }
+
+  async openIntentApp(appId): Promise<AppIdentifier> {
+    try {
+      return await fdc3.open({ appId: appId });
+    } catch (ex) {
+      assert.fail(`Error while attempting to open the mock app: ${ex.message ?? ex}`);
+    }
+  }
+
+  async createAppChannel(channelId: string): Promise<Channel> {
+    return await fdc3.getOrCreateChannel(channelId);
+  }
+
+  async createPrivateChannel(): Promise<PrivateChannel> {
+    return await fdc3.createPrivateChannel();
+  }
+
+  validatePrivateChannel(privChan: PrivateChannel): void {
+    expect(privChan).to.have.property("id");
+  }
+
+  async raiseIntent(intent: string, contextType: string, appIdentifier?: AppIdentifier, delayBeforeReturn: number = 0, contextId?: { [key: string]: string }): Promise<IntentResolution> {
+    let context: IntentUtilityContext = {
+      type: contextType,
+      delayBeforeReturn: delayBeforeReturn,
+    };
+
+    if (contextId) {
+      context.id = contextId;
+    }
+
+    try {
+      if (appIdentifier) {
+        return await fdc3.raiseIntent(intent, context, appIdentifier);
+      } else {
+        return await fdc3.raiseIntent(intent, context);
+      }
+    } catch (ex) {
+      throw new Error(`${ex.message ?? ex}`);
+    }
+  }
+
+  async findInstances(appId: string): Promise<AppIdentifier[]> {
+    try {
+      return await fdc3.findInstances({ appId: appId });
+    } catch (ex) {
+      assert.fail(`Error while attempting to find instances: ${ex.message ?? ex}`);
+    }
+  }
+
+  getIntentResult(intentResolution: IntentResolution): Promise<IntentResult> {
+    //ensure getIntentResult immediately returns a promise that can be awaited
+    let timeout = this.failIfPromiseNotReceived();
+    let intentResult = intentResolution.getResult().catch((ex) => {
+      assert.fail(`Error while attempting to retrieve the IntentResult from the IntentResolution object: ${ex.message ?? ex}`);
+    });
+    clearTimeout(timeout);
+    return intentResult;
+  }
```

Comment:
Because you've got a `.catch` on the `.getResult()`, it will eat the promise when an error occurs. You'll get a failure on the test (which may be correct, although not in every test, some want to see the errors), you then return void instead of a promise, which in turn causes an uncaught error in the test.

I think a better way to go would be something like:

```typescript
getIntentResult(intentResolution: IntentResolution): Promise<IntentResult> {
    //ensure getIntentResult immediately returns a promise that can be awaited
    const timeout = this.failIfResponseTimesOut();
    const intentResult = intentResolution.getResult();
    if (typeof intentResult.then !== "function"){
      assert.fail(`intentResolution.getResult() did not return a Promise: ${JSON.stringify(intentResult, null, 2)}`);
      return null;
    }
    clearTimeout(timeout);
    return intentResult;
  }

  failIfResponseTimesOut() {
    let timeout = window.setTimeout(() => {
      assert.fail("When running getIntentResult() the promise should be returned immediately unless it is being awaited");
    }, 500);

    return timeout;
  }
  ```
  
then check in the actual test whether you got an error or the desired response (as that varies by test).


Hence, the test becomes:
```typescript
    const RaiseIntentVoidResult0secs = "(2.0-RaiseIntentVoidResult0secs) App A receives a void IntentResult";
    it(RaiseIntentVoidResult0secs, async () => {
      await control.listenForError();
      const intentResolution = await control.raiseIntent("aTestingIntent", "testContextX");
      control.validateIntentResolution("IntentAppAId", intentResolution);
      let intentResultPromise = control.getIntentResult(intentResolution);
      if (intentResultPromise) {
          const intentResult = await intentResultPromise;
          control.validateIntentResult(intentResult, IntentResultType.Void);
      } // will have already failed test... although might be worth having another explicit fail here
    });
```